radial_build_tools.py

```diff
@@ -0,0 +1,474 @@
+            boundary_type="vacuum",
+        )
+
+        vac_region = -vac_surf & +self.surfaces[self.surf_list[-1]]
+        vac_cell = openmc.Cell(region=vac_region, name="vac_cell")
+
+        self.cell_list.append(vac_cell)
+        self.cell_dict["vac_cell"] = vac_cell
+
+        self.geometry = openmc.Geometry(self.cell_list)
+
+    def build_openmc_model(self):
+        """
+        Builds openmc model using the build definition
+        """
+        self.build_surfaces()
+        self.build_regions()
+        self.build_cells()
+        self.get_bounded_geometry()
+
+    def get_openmc_model(self):
+        """
+        Return toroidal model built using the build definition, contains
+        geometry and material information
+
+        Returns:
+            model (openmc model): Model containing materials and geometry
+                from the build dict.
+            cells (dict): dict mapping layer names to openmc cell instances in
+                the model object returned by this function.
+        """
+        self.build_openmc_model()
+        model = openmc.Model(geometry=self.geometry, materials=self.materials)
+        return model, self.cell_dict
+
+    def get_radial_build_plot(
+        self,
+        title="radial_build",
+        colors=None,
+        max_characters=35,
+        max_thickness=1e6,
+        size=(8, 4),
+        unit="cm",
+    ):
+        """
+        Make a radial build plot object using the build dictionary of this model.
+
+        Arguments:
+            title (string): title for plot and filename to save to
+            colors (list of str): list of matplotlib color strings.
+                If specific colors are desired for each layer they can be added
+                here.
+            max_characters (float): maximum length of a line before wrapping the
+                text
+            max_thickness (float): maximum thickness of layer to display, useful
+                for reducing the total size of the figure.
+            size (iter of float): figure size, inches. (width, height)
+            unit (str): Unit of thickness values
+
+        Returns:
+            RadialBuildPlot (obj): radial build plot object.
+        """
+        return RadialBuildPlot(
+            self.build,
+            title=title,
+            colors=colors,
+            max_characters=max_characters,
+            max_thickness=max_thickness,
+            size=size,
+            unit=unit,
+        )
+
+
+def parse_args():
+    """Parser for running as a script"""
+    parser = argparse.ArgumentParser(prog="plot_radial_build")
+
+    parser.add_argument("filename", help="YAML file defining radial build")
+
+    return parser.parse_args()
+
+
+def read_yaml(filename):
+    """Reads yaml file to extract title and build variables"""
+    with open(filename) as file:
+        data = yaml.safe_load(file)
+
+    return data
+
+
+def main():
+
+    # default data for running from command line
+    data_default = {
```

Comment:
We switched to a different way of handling defaults in parastell based on it now being an object.  That might work well here, too.

Reply:
switched to using kwargs to handle the default values for `RadialBuildPlot`